### PR TITLE
Accounts updated subscription

### PR DESCRIPTION
**Pull request by Victor-N-Suadicani**
## Purpose

I'm honestly not sure where this is used in the frontend.

## Changes

Added the `AccountsUpdatedSubscriptionItem` and the appropriate methods to `Subscription`, which are largely inspired by the already-existing `block_added` subscription.

What's missing is actually sending the notifications when the appropriate event happens - this is left for future work.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by DOBEN** [comment]
The `AccountsUpdatedSubscriptionItem` is used as part of the web socket subscription when the accounts page view is open e.g.
https://testnet.ccdscan.io/?dcount=1&dentity=account&daddress=4dT5vPrnnpwVrXZgmYLtHrDLvBYhtzheaK4fDWbJewqRCGQKWz

In dev tools view open the `network` tab and click on the `ws` tab to explore the web socket connections.

File: backend-rust/src/graphql_api.rs [hunk 1]
```diff
@@ -997,42 +998,56 @@ LIMIT 30", // WHERE slot_time > (LOCALTIMESTAMP - $1::interval)
...
 impl Subscription {
-    async fn block_added(
-        &self,
-    ) -> impl Stream<Item = Result<Arc<Block>, tokio_stream::wrappers::errors::BroadcastStreamRecvError>>
-    {
+    async fn block_added(&self) -> impl Stream<Item = Result<Block, BroadcastStreamRecvError>> {
         tokio_stream::wrappers::BroadcastStream::new(self.block_added.resubscribe())
     }
+
+    async fn accounts_updated(
```
Comment (DOBEN) [inline backend-rust/src/graphql_api.rs:1028]: The query at the front-end includes an additional `accountAddress` query variable:
```

{
    "id": "1",
    "type": "start",
    "payload": {
        "key": "...",
        "query": "subscription accountsUpdatedSubscription($accountAddress: String!) {\n  accountsUpdated(accountAddress: $accountAddress) {\n    address\n  }\n}",
        "variables": {
            "accountAddress": "4dT5vPrnnpwVrXZgmYLtHrDLvBYhtzheaK4fDWbJewqRCGQKWz"
        },
        "context": {
            "url": "https://api-ccdscan.testnet.concordium.com/graphql",
            "preferGetMethod": false,
            "suspense": false,
            "requestPolicy": "cache-first",
            "meta": {
                "cacheOutcome": "miss"
            }
        }
    }
}
```

Comment (Victor-N-Suadicani) [reply backend-rust/src/graphql_api.rs:1028]: Right - but what am I supposed to do with that input string? :thinking:

Comment (DOBEN) [reply backend-rust/src/graphql_api.rs:1028]: The current front-end allows to pass in an `accountAddress` (or the empty string) as a variable:
https://github.com/Concordium/concordium-scan/blob/main/frontend/src/components/Accounts/AccountDetailsContainer.vue#L120

A meaningful logic would be that we either simplify it and not pass a variable here, the notification would be triggered whenever a new row is added to the `affected_accounts` table (option 1).
The front-end uses the following logic anyway:
https://github.com/Concordium/concordium-scan/blob/main/frontend/src/components/Accounts/AccountDetailsContainer.vue#L107

or we keep the variable (require even the variable), the notification would be triggered whenever a new row is added to the `affected_accounts` table and the `affected_account` equals the variable.

Comment (DOBEN) [reply backend-rust/src/graphql_api.rs:1055]: I chose to implement `option 2` now.

**Comment by DOBEN** [comment]
How to test:
- Open `http://localhost:8000/` in browser
- Add subscription:
```
subscription accountsUpdatedSubscription($accountAddress: String) {
  accountsUpdated(accountAddress: $accountAddress) {
    address
  }
}
```
- Add variable (optional): e.g.
```
{
  "accountAddress": "4Y1c27ZRpRut9av69n3i1uhfeDp4XGuvsm9fkEjFvgpoxXWxQB"
}

```
- Press `Play` button:
- Obesrve `Subscriptions` coming in on the right side.

Screenshot from 2024-11-29 18-02-24

**Review by soerenbf — COMMENTED** [review]
Still need to test it, but here are some initial observations

**Review by soerenbf — APPROVED** [review]
Managed to subscribe and receive an update 🎉

File: backend-rust/migrations/0001_initialize.up.sql [hunk 2]
```diff
@@ -373,6 +373,32 @@ CREATE INDEX event_index_per_contract_idx ON contract_events (event_index_per_co
 -- Important for quickly filtering contract events by a specific contract.
 CREATE INDEX contract_events_idx ON contract_events (contract_index, contract_sub_index);
 
+CREATE OR REPLACE FUNCTION account_updated_notify_trigger_function() RETURNS trigger AS $trigger$
+DECLARE
+  rec affected_accounts;
+  payload TEXT;
+  lookup_result TEXT;
```
[the remark this answers is not in the document]
Comment (DOBEN) [reply backend-rust/migrations/0001_initialize.up.sql:380]: eagle eyes :)

I removed the unused `payload`.

File: backend-rust/src/graphql_api.rs [hunk 3]
```diff
@@ -1024,42 +1025,85 @@ LIMIT 30", // WHERE slot_time > (LOCALTIMESTAMP - $1::interval)
...
+                                Some(Ok(notification))
+                            }
+                        }
+                        // Pass on errors.
+                        Err(e) => Some(Err(e)),
+                    }
+                }
+            },
+        )
+    }
```
[the remark this answers is not in the document]
Comment (DOBEN) [reply backend-rust/src/graphql_api.rs:1088]: The `filter_map` function expects a `Future` and the address couldn't be moved into the `Future`.

**Pull request merged**